workers/common/src/test/kotlin/common/env/GradleInitGeneratorTest.kt

```diff
@@ -48,6 +48,312 @@ class GradleInitGeneratorTest : WordSpec({
             mockBuilder.homeFileName shouldBe ".gradle/init.gradle.kts"
         }
 
+        "generate repositories blocks from definitions" {
+            val secUser1 = MockConfigFileBuilder.createSecret("user1Secret")
+            val secPass1 = MockConfigFileBuilder.createSecret("pass1Secret")
+            val secUser2 = MockConfigFileBuilder.createSecret("user2Secret")
+            val secPass2 = MockConfigFileBuilder.createSecret("pass2Secret")
+
+            val definition1 = GradleDefinition(
+                MockConfigFileBuilder.createInfrastructureService("https://repo1.example.org", secUser1, secPass1),
+                emptySet()
+            )
+
+            val definition2 = GradleDefinition(
+                MockConfigFileBuilder.createInfrastructureService("https://repo2.example.org", secUser2, secPass2),
+                emptySet()
+            )
+
+            val mockBuilder = MockConfigFileBuilder()
+
+            GradleInitGenerator().generate(mockBuilder.builder, listOf(definition1, definition2))
+
+            val expectedLines = listOf(
```

Comment:
How about beautifying this with the

```kotlin
"""
...
""".trimIndent()
```

approach?

Reply:
FYI, I've done this now also for existing code in https://github.com/eclipse-apoapsis/ort-server/pull/3153.